ngtools/webpack overwrites special webpack's request

<!--
IF YOU DON'T FILL OUT THE FOLLOWING INFORMATION YOUR ISSUE MIGHT BE CLOSED WITHOUT INVESTIGATING
-->
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Area
```
- [x] devkit
- [ ] schematics
```

### Versions
<!--
Output from: `node --version` and `npm --version`.
  Windows (7/8/10). Linux (incl. distribution). macOS (El Capitan? Sierra? High Sierra?)
-->

```
"@angular/compiler-cli": "5.2.10",
"@ngtools/webpack": "1.10.2",
npm - 4.1.2
node - 8.9.3
```

### Repro steps
[Repository with reproduction](https://github.com/WiseBird/ngtools_tsconfig_path_wildcard) (run `yarn install && yarn run build`)
<!--
Simple steps to reproduce this bug.
Please include: commands run, packages added, related code changes.
A link to a sample repo would help too.
-->
* Use external library with `define` - [brace](https://github.com/thlorenz/brace) in my case
* Add to `tsconfig.json` catch all path:
```
    "paths": {
      "*": [
        "./node_modules/*"
      ]
    },
```

### The log given by the failure
```
ERROR in ./node_modules/brace/index.js
Module not found: Error: Can't resolve '.../node_modules/' in '.../node_modules/brace'
 @ ./node_modules/brace/index.js 4181:93-99 20208:21-27
 @ ./src/app/root/app.component.ts
 @ ./src/app/app.module.ngfactory.js
 @ ./src/index.ts
``` 


### Desired functionality
No error


### Mention any other details that might be useful
When encountering `define` in js file webpack issues special request: `!!webpack amd define`. `ngtools` wrongly rewrites it to `node_modules/!!webpack amd define` (in `@ngtools/webpack/src/paths-plugin.js:resolveWithPaths`).

--- a/packages/ngtools/webpack/src/paths-plugin.ts
+++ b/packages/ngtools/webpack/src/paths-plugin.ts
@@ -42,6 +42,13 @@ export function resolveWithPaths(
     return;
   }
 
+  // Amd requests are not mapped
+  if (originalRequest.startsWith('!!webpack amd')) {
+    callback(null, request);
+
+    return;
+  }
+
   // check if any path mapping rules are relevant
   const pathMapOptions = [];
   for (const pattern in compilerOptions.paths) {

